nRFMeshProvision/MeshNetworkManager.swift

```diff
@@ -276,8 +265,8 @@ public extension MeshNetworkManager {
         guard let networkManager = networkManager else {
             return
         }
-        queue.async {
-            networkManager.handle(incomingPdu: data, ofType: type)
+        Task {
```

Comment:
I'm actually considering removing `async` from `handle(incomingPdu:ofType:)` in `NetworkManager`. It's not asynchronous anyway.
I'd just call it from `Task.detached` in `MeshNetworkManager`, but the `bearerDidDeliverData` can be non-blocking only.

Reply:
Agree - ultimately want a synchronous call from NetworkConnection/BluetoothManager thread to quickly deposit incoming msg in a receive buffer so that thread/task waiting doesn't cause msg reordering. Unclear how to best architect this until start introducing actors (wouldn't want to block while waiting for a networking stack actor to complete something). Since AccessLayer is out of scope of this chg, didn't make the comment yesterday.